tutorial/server/actix_web/src/main.rs

```diff
@@ -0,0 +1,56 @@
+use actix_session::storage::CookieSessionStore;
+use actix_session::SessionMiddleware;
+use actix_web::cookie::Key;
+use actix_web::middleware::Logger;
+use actix_web::web::JsonConfig;
+use actix_web::web::{get, post};
+use actix_web::{App, HttpServer};
+use log::info;
+
+use crate::handler::auth::{
+    finish_authentication, finish_register, start_authentication, start_register,
+};
+use crate::handler::index::index;
+use crate::handler::serve_wasm::serve_wasm;
+use crate::startup::startup;
+
+mod handler;
+mod startup;
+
+#[tokio::main]
+async fn main() {
+    // Initialize env-logger
+    env_logger::init();
+
+    // Generate secret key for cookies.
+    // Normally you would read this from a configuration file.
+    let key = Key::generate();
+
+    let (webauthn, webauthn_users) = startup();
+
+    // Build the webserver and run it
+    info!("Start listening on: http://0.0.0.0:8080");
+    HttpServer::new(move || {
+        App::new()
+            .wrap(Logger::default())
+            .wrap(
+                SessionMiddleware::builder(CookieSessionStore::default(), key.clone())
```

Comment:
I think the cookie session store here stores the full state in the cookie itself, rather that being a key to a memory map. this means in this example replay attacks are possible, which is not great in an example. 

So I think you may need either add a server-side memory session store in actix_session, or you need to add a hashmap or similar to the app state, and then store a key-id or similar in the cookie instead. Thoughts?

Reply:
Implemented in d72ab9702768ce14ba8b3d0d10422986ae92c3d0